### PR TITLE
fix(ci): Correct files-changed variable name 

### PR DESCRIPTION
**Pull request by armenzg**
This is very easy to regress.
This happened in https://github.com/getsentry/sentry/pull/38881
This means that we have been carrying forward old coverage data from four months ago for the acceptance and charcuterie jobs.

### CODE REVIEW

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -122,7 +115,7 @@ jobs:
       # it reduces large coverage fluctuations
       - name: Handle artifacts
         uses: ./.github/actions/artifacts
-        if: ${{ always() && needs.files-changed.outputs.frontend_all == 'true' }}
+        if: always() && needs.files-changed.outputs.frontend_all == 'true'
```
Comment (armenzg) [inline .github/workflows/acceptance.yml:118]: Okay. This does not work. Let's go back.
image

File: .github/workflows/acceptance.yml [hunk 2]
```diff
@@ -31,7 +31,8 @@ jobs:
     # Map a step output to a job output
     outputs:
       acceptance: ${{ steps.changes.outputs.acceptance }}
-      backend: ${{ steps.changes.outputs.backend_all }}
+      backend_all: ${{ steps.changes.outputs.backend_all }}
```
Comment (armenzg) [inline .github/workflows/acceptance.yml:34]: You can see that we were condition on `backend_all` but only `backend` would be available before this.
https://github.com/getsentry/sentry/blob/b9ed360540ae6d31bdc827a500ec8f761ead5fd4/.github/workflows/acceptance.yml#L215
https://github.com/getsentry/sentry/blob/b9ed360540ae6d31bdc827a500ec8f761ead5fd4/.github/workflows/acceptance.yml#L289

File: .github/workflows/acceptance.yml [hunk 3]
```diff
@@ -31,7 +31,8 @@ jobs:
...
+      frontend_all: ${{ steps.changes.outputs.frontend_all }}
```
Comment (armenzg) [inline .github/workflows/acceptance.yml:35]: `frontend_all` was only added in the last week, thus, the drift was not as bad
https://github.com/getsentry/sentry/blob/b9ed360540ae6d31bdc827a500ec8f761ead5fd4/.github/workflows/acceptance.yml#L126

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**